liblangutil/SemVerHandler.cpp

```diff
@@ -171,14 +172,20 @@ optional<SemVerMatchExpression> SemVerMatchExpressionParser::parse()
 			if (m_pos >= m_tokens.size())
 				break;
 			if (currentToken() != Token::Or)
-				BOOST_THROW_EXCEPTION(SemVerError());
+			{
+				string tokenString = TokenTraits::toString(currentToken());
+				BOOST_THROW_EXCEPTION(
+					SemVerError() << errinfo_comment("Invalid token or "
+					"more than one version not delimited by || token: "s + tokenString)
```

Comment:
This message should be more precise. As is, it may be confusing because normally we do not require `||` between versions - if you do not use it it's just interpreted as `&&` instead and it's perfectly fine. The one situation where that's not the case is a range of versions (e.g. `0.8.3 - 0.8.17`). Looking at `parseMatchExpression()`, I see we can have multiple versions next to each other but not multiple ranges. Or a range next to a version.

Not sure how to phrase it though. This would be precise, though it still does not explain why:
```c++
"Invalid token: "s + tokenString + ". Note that conjunction is not allowed at this position."
```

Maybe we can come up with something better.

To be honest, this is a weird limitation. I don't see why conjunction would have to be disallowed here.

Reply:
Please also add a test case covering this. Currently we don't seem to have any.

Reply:
So, after discussing this with @matheusaaguiar and @nikola-matic, let's make the message here something like "You can only combine version ranges using the || operator."